qmf/src/plugins/messageservices/imap/imapstrategy.cpp

```diff
@@ -155,13 +155,14 @@ bool updateMessagesMetaData(ImapStrategyContextBase *context,
                                    const QMailMessageKey &storedKey, 
                                    const QMailMessageKey &unseenKey, 
                                    const QMailMessageKey &seenKey,
+                                   const QMailMessageKey &flaggedAsDeleted,
                                    const QMailMessageKey &flaggedKey,
                                    const QMailMessageKey &unreadElsewhereKey,
                                    const QMailMessageKey &importantElsewhereKey,
                                    const QMailMessageKey &unavailableKey)
 {
     bool result = true;
-    QMailMessageKey reportedKey(seenKey | unseenKey);
+    QMailMessageKey reportedKey((seenKey | unseenKey)  & ~flaggedAsDeleted);
```

Comment:
I have no hope of following this without watching the protocol, I'll take your word for it :)